pkgs/development/compilers/cudatoolkit/flags.nix

```diff
@@ -13,66 +24,120 @@ let
   #
   # Please see the accompanying documentation or https://github.com/NixOS/nixpkgs/pull/205351
 
-  defaultCudaCapabilities = rec {
-    cuda9 = [
-      "3.0"
-      "3.5"
-      "5.0"
-      "5.2"
-      "6.0"
-      "6.1"
-      "7.0"
-    ];
-
-    cuda10 = cuda9 ++ [
-      "7.5"
-    ];
-
-    cuda11 = [
-      "3.5"
-      "5.0"
-      "5.2"
-      "6.0"
-      "6.1"
-      "7.0"
-      "7.5"
-      "8.0"
-      "8.6"
-    ];
-
-  };
-
-  cudaMicroarchitectureNames = {
-    "3" = "Kepler";
-    "5" = "Maxwell";
-    "6" = "Pascal";
-    "7" = "Volta";
-    "8" = "Ampere";
-    "9" = "Hopper";
-  };
-
-  defaultCudaArchList = defaultCudaCapabilities."cuda${lib.versions.major cudatoolkit.version}";
-  cudaRealCapabilities = config.cudaCapabilities or defaultCudaArchList;
-  capabilitiesForward = "${lib.last cudaRealCapabilities}+PTX";
-
-  dropDot = ver: builtins.replaceStrings ["."] [""] ver;
-
-  archMapper = feat: map (ver: "${feat}_${dropDot ver}");
-  gencodeMapper = feat: map (ver: "-gencode=arch=compute_${dropDot ver},code=${feat}_${dropDot ver}");
-  cudaRealArchs = archMapper "sm" cudaRealCapabilities;
-  cudaPTXArchs = archMapper "compute" cudaRealCapabilities;
-  cudaArchs = cudaRealArchs ++ [ (lib.last cudaPTXArchs) ];
-
-  cudaArchNames = lib.unique (map (v: cudaMicroarchitectureNames.${lib.versions.major v}) cudaRealCapabilities);
-  cudaCapabilities = cudaRealCapabilities ++ lib.optional (config.cudaForwardCompat or true) capabilitiesForward;
-  cudaGencode = gencodeMapper "sm" cudaRealCapabilities ++ lib.optionals (config.cudaForwardCompat or true) (gencodeMapper "compute" [ (lib.last cudaPTXArchs) ]);
-
-  cudaCapabilitiesCommaString = lib.strings.concatStringsSep "," cudaCapabilities;
-  cudaCapabilitiesSemiColonString = lib.strings.concatStringsSep ";" cudaCapabilities;
-  cudaRealCapabilitiesCommaString = lib.strings.concatStringsSep "," cudaRealCapabilities;
+  # gpus :: List Gpu
+  gpus = trivial.importJSON ./gpus.json;
```

Comment:
I think of JSON as a format for storing data more than I do nix, but there's no reason it can't be a nix file.

Would you prefer a standalone file or for it to be in the same file? My concern with it being in the same file is that it may get very long.

Reply:
Yeah, Nix (the language) is also a config/data format like JSON, although it's easy to lose sight of it. There are a few JSON files in nixpkgs (eg cudaPackages) but this is almost always because they are downloaded from some upstream source. In this case this file is being written by hand, so let's stick with Nix.

> Would you prefer a standalone file or for it to be in the same file? My concern with it being in the same file is that it may get very long.

Agreed, let's keep it to a separate file!

Reply:
Done in new commit

Reply:
The main difference being that JSON is easily manipulated with tooling, and that is not so much the case for Nix.